src/builder.rs

```diff
@@ -206,14 +216,19 @@ pub enum InitError<DI, P> {
     /// This error is returned when the configuration passed to the builder is
     /// invalid. For example, when the combination of bit depth and interface
     /// kind isn't supported by the selected model.
-    InvalidConfiguration,
+    InvalidConfiguration(&'static str),
```

Comment:
I did think about this some more and I came to the conclusion that an enum would work better than a string. By using an enum we wouldn't have to store the error strings in flash and we could instead add doc comments to the enum variants to provide better information to the user. By making the enum `non_exhaustive` we would still be flexible and could always add new error kinds without causing a breaking change.

We might still want to have a string variant to allow external `Model` impls to report other kinds of errors.

Something like this:
```rust
pub enum InitError<...> {
   ...,
   InvalidConfiguration(pub ConfigurationError),
}

#[non_exhaustive]
pub enum ConfigurationError {
   InvalidDisplaySize,
   InvalidColorFormat,
   Other(&'static str),
}
```

Reply:
refactored, good point about taking the string memory bytes.